src/testing/snaptest.zig

```diff
@@ -0,0 +1,220 @@
+        if (std.mem.eql(u8, got, snapshot.text)) return;
+
+        std.debug.print(
+            \\Snapshot differs.
+            \\Want:
+            \\----
+            \\{s}
+            \\----
+            \\Got:
+            \\----
+            \\{s}
+            \\----
+            \\
+        ,
+            .{
+                snapshot.text,
+                got,
+            },
+        );
+
+        if (!snapshot.should_update()) {
+            return error.SnapDiff;
+        }
+
+        var arena = std.heap.ArenaAllocator.init(std.testing.allocator);
+        defer arena.deinit();
+
+        const allocator = arena.allocator();
+
+        const file_text =
+            try std.fs.cwd().readFileAlloc(allocator, snapshot.location.file, 1024 * 1024);
+        var file_text_updated = try std.ArrayList(u8).initCapacity(allocator, file_text.len);
+
+        const line_zero_based = snapshot.location.line - 1;
+        const range = snap_range(file_text, line_zero_based) orelse
+            return error.SnapNotFound;
+
+        const snapshot_prefix = file_text[0..range.start];
+        const snapshot_text = file_text[range.start..range.end];
+        const snapshot_suffix = file_text[range.end..];
+
+        const indent = get_indent(snapshot_text);
+
+        try file_text_updated.appendSlice(snapshot_prefix);
+        {
+            var lines = std.mem.split(u8, got, "\n");
+            while (lines.next()) |line| {
+                try file_text_updated.writer().print("{s}\\\\{s}\n", .{ indent, line });
+            }
+        }
+        try file_text_updated.appendSlice(snapshot_suffix);
+
+        try std.fs.cwd().writeFile(snapshot.location.file, file_text_updated.items);
+
+        std.debug.print("Updated {s}\n", .{snapshot.location.file});
+        return error.SnapUpdated;
+    }
+};
+
+const Range = struct { start: usize, end: usize };
+fn snap_range(text: []const u8, snap_line: u32) ?Range {
+    var offset: usize = 0;
+    var line_number: u32 = 0;
+
+    var lines = std.mem.split(u8, text, "\n");
+    const snap_start = while (lines.next()) |line| : (line_number += 1) {
+        if (line_number == snap_line) {
+            assert(std.mem.indexOf(u8, line, "@src()") != null);
+        }
+        if (line_number == snap_line + 1) {
+            assert(is_multiline_string(line));
+            break offset;
+        }
+        offset += line.len + 1; // 1 for \n
+    } else return null;
+
+    lines = std.mem.split(u8, text[snap_start..], "\n");
+    const snap_end = while (lines.next()) |line| {
+        if (!is_multiline_string(line)) {
+            break offset;
+        }
+        offset += line.len + 1; // 1 for \n
+    } else return null;
+
+    return Range{ .start = snap_start, .end = snap_end };
+}
+
+fn is_multiline_string(line: []const u8) bool {
+    for (line) |c, i| {
+        switch (c) {
+            ' ' => {},
+            '\\' => return (i < line.len and line[i + 1] == '\\'),
```

Comment:
Should the first clause be `i + 1 < line.len`?

Reply:
Indeed! In Rust, I'd spell this as `line[i..].starts_with(r"\\")` to be more robust, but in Zig `std.mem.startsWith(u8, ` feels like a mouthful and doesn't feel minimal enough.